server/webapp/WEB-INF/rails.new/app/presenters/api_v1/command_snippet_representer.rb

```diff
@@ -0,0 +1,45 @@
+##########################################################################
+# Copyright 2016 ThoughtWorks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##########################################################################
+
+module ApiV1
+  class CommandSnippetRepresenter < ApiV1::BaseRepresenter
+    alias_method :command_snippet, :represented
+    link :self do |opts|
+      opts[:url_builder].apiv1_admin_command_snippet_url(relative_path: command_snippet.getRelativePath())
+    end
+
+    link :doc do |opts|
+      'http://api.go.cd/#command_snippets'
+    end
+
+    link :find do |opts|
+      opts[:url_builder].apiv1_admin_command_snippet_url(':relative_path')
+    end
+
+    property :getName, as: :name
+    property :getDescription, as: :description
+    property :getAuthor, as: :author
+    property :getAuthorInfo, as: :author_info
+    property :getMoreInfo, as: :more_info
+    property :getCommandName, as: :command
+    property :arguments, exec_context: :decorator
+    property :getRelativePath, as: :relative_path
```

Comment:
Not sure I follow, let's talk about it on Monday

On Sat, Jan 23, 2016, 5:02 PM Mahesh Panchaksharaiah <
notifications@github.com> wrote:

> In
> server/webapp/WEB-INF/rails.new/app/presenters/api_v1/command_snippet_representer.rb
> <https://github.com/gocd/gocd/pull/1823#discussion_r50619292>:
>
> > +    link :doc do |opts|
> > +      'http://api.go.cd/#command_snippets'
> > +    end
> > +
> > +    link :find do |opts|
> > +      opts[:url_builder].apiv1_admin_command_snippet_url(':relative_path')
> > +    end
> > +
> > +    property :getName, as: :name
> > +    property :getDescription, as: :description
> > +    property :getAuthor, as: :author
> > +    property :getAuthorInfo, as: :author_info
> > +    property :getMoreInfo, as: :more_info
> > +    property :getCommandName, as: :command
> > +    property :arguments, exec_context: :decorator
> > +    property :getRelativePath, as: :relative_path
>
> Currenty the CommandRepositoryService has only one api to fetch the
> CommandSnippet which is getCommandSnippetByRelativePath, thats the reason
> for this endpoint
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gocd/gocd/pull/1823/files#r50619292>.
>